backend/inspirehep/records/api/mixins.py

```diff
@@ -381,7 +390,50 @@ def update(self, data, disable_relations_update=False, *args, **kwargs):
             self.update_conference_paper_and_proccedings()
         else:
             LOGGER.info(
-                "Record conference papaers and proccedings update disabled",
+                "Record conference papers and proceedings update disabled",
                 recid=self.get("control_number"),
                 uuid=str(self.id),
             )
+
+    def get_updated_referenced_conference(self):
+        try:
+            prev_version = self._previous_version
+        except AttributeError:
+            prev_version = {}
+
+        changed_deleted_status = self.get("deleted", False) ^ prev_version.get(
+            "deleted", False
+        )
+
```

Comment:
I would check `changed_deleted_status` already here and if `True` loads `pids_latest` and returns, without doing all the other operations